extensions/eclipse-che-theia-user-preferences/src/node/che-theia-preferences-contribution.ts

```diff
@@ -0,0 +1,31 @@
+/********************************************************************************
+ * Copyright (C) 2018-2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from "inversify";
+import { BackendApplicationContribution } from '@theia/core/lib/node/backend-application';
+import { CheTheiaUserPreferencesSynchronizer } from './che-theia-preferences-synchronizer';
+
+@injectable()
+export class CheTheiaPreferencesContribution implements BackendApplicationContribution {
+
+    @inject(CheTheiaUserPreferencesSynchronizer)
+    cheTheiaUserPreferencesSynchronizer: CheTheiaUserPreferencesSynchronizer
+
+    async onStart(): Promise<void> {
+        await this.cheTheiaUserPreferencesSynchronizer.readTheiaUserPreferencesFromCheSettings();
```

Comment:
is it an issue to block the start of theia to wait for the preferences ?

Reply:
The idea was to provide preferences before Theia start.
I do not mind to remove await here, watcher should do the job.